### ISSUE TITLE
What's the purpose of `Ensemble.predictions`?

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/pyPESTO#1357]
I don't quite understand the purpose of `Ensemble.predictions`. Does anybody know what this was intended for? I don't find it really being used anywhere.

https://github.com/ICB-DCM/pyPESTO/blob/b50bc8fc1853002d845db80f7e2332cb410f92a6/pypesto/ensemble/ensemble.py#L509-L510

I would find it more plausible if some predictions would store a reference to the ensemble they were generated from, rather than the other way around.

**Comment by PaulJonasJost** [comment]
Agreed, quickly went through the codebase and it seems to be not used unless to allow functions that use `get_prediction_dataset` to be called with an ensemble. Should be changed. Fine with removing and would do it if others also agree. I think this is confusing. Would also add the ensemble as reference to the `EnsemblePRediction`.
